packages/fether-react/src/Accounts/CreateAccount/AccountImportOptions/AccountImportOptions.js

```diff
@@ -0,0 +1,156 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+import React, { Component } from 'react';
+import { Card, Form as FetherForm } from 'fether-ui';
+import { inject, observer } from 'mobx-react';
+
+@inject('createAccountStore')
+@observer
+class AccountImportOptions extends Component {
+  state = {
+    error: '',
+    isLoading: false,
+    isFileValid: false,
+    json: null,
+    value: ''
+  };
+
+  handleNextStep = async () => {
+    const {
+      history,
+      location: { pathname },
+      createAccountStore: { isImport, isJSON, setJSON, setPhrase }
+    } = this.props;
+    const currentStep = pathname.slice(-1);
+    const { json, value } = this.state;
+
+    if (isJSON) {
+      this.setState({ isLoading: true });
+      await setJSON(json);
+    }
+
+    if (isImport && !isJSON) {
+      this.setState({ isLoading: true });
+      await setPhrase(value);
+    }
+
+    history.push(`/accounts/new/${+currentStep + 1}`);
+  };
+
+  handleChange = ({ target: { value } }) => {
+    this.setState({ value });
+  };
+
+  handleChangeFile = data => {
+    const {
+      createAccountStore: { setIsJSON }
+    } = this.props;
+
+    try {
+      const json = JSON.parse(data);
+
+      const isFileValid =
+        json && json.address.length === 40 && json.version === 3;
+
+      if (isFileValid) {
```

Comment:
Shouldn't you throw an error here if file is not valid? Or else `handleNextStep` will be called if file is not valid.

Reply:
this is true. though handleNextStep is in the if block so nothing would happen at all. Either way, needs to be fixed.